src/formatter.rs

```diff
@@ -182,7 +186,15 @@ impl<'a> Formatter<'a> {
         {
             self.trim_spaces_end(query);
         }
-        if self.options.uppercase {
+        if self.options.uppercase
+            && (self.options.ignore_case_convert.is_none()
+                || !self
+                    .options
+                    .ignore_case_convert
+                    .as_ref()
+                    .unwrap()
+                    .contains(&token.value))
```

Comment:
It's probably more idiomatic to do this with a `map` (apply to 2 more instances below as well)

```suggestion
            && !self
                    .options
                    .ignore_case_convert
                    .as_ref()
                    .map(|values| values.contains(&token.value))
                    .unwrap_or(false)
```